CMakeLists.txt

```diff
@@ -93,6 +93,11 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR
 
 	# Disable warnings that produce more headaches than use.
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-parameter")
+
+	# Add some libraries needed for the MSYS build.
+	if(MSYS)
+		set(CMAKE_CXX_STANDARD_LIBRARIES "-lshlwapi -lz ${CMAKE_CXX_STANDARD_LIBRARIES}")
```

Comment:
From CMake's documentation on [`CMAKE_CXX_STANDARD_LIBRARIES`](https://cmake.org/cmake/help/v3.0/variable/CMAKE_STANDARD_LIBRARIES.html):
> Libraries linked into every executable and shared library.

Are you sure that these two libraries need to be linked into **every** executable and shared library? For example, when looking which parts of RetDec use `shlwapi.h`, I have found only `retdec-utils`:
```
$ git grep shlwapi\.h
src/utils/filesystem_path.cpp:  #include <shlwapi.h>
```
The recommended way would be to add this library via `target_link_libraries()` in `src/utils/CMakeLists.txt`. Since it is already there for MSVC, I suggest simply changing
```
if(MSVC)
    target_link_libraries(retdec-utils shlwapi) # shlwapi.dll for PathRemoveFileSpec()
endif()
```
to
```
if(MSVC OR MSYS)
    target_link_libraries(retdec-utils shlwapi) # shlwapi.dll for PathRemoveFileSpec()
endif()
```

The second library is `libz` (`-lz`). AFAIK the only part that uses `libz` is LLVM. Thus, I would suggest trying adding that dependency into `deps/llvm/CMakeLists.txt` after line 226 (you can use `elseif` if you want).
```
if(MSYS)
    target_link_libraries(llvm INTERFACE debug     ${ZLIB_LIBRARIES})
    target_link_libraries(llvm INTERFACE optimized ${ZLIB_LIBRARIES})
endif()
```

Reply:
I'm primary developer of MSYS2 and I can say that condition in patch is wrong.
MSYS under CMAKE mean that you use MSYS gcc and not MINGW GCC.

Patch must be modified to:
````
if(WIN32)
    target_link_libraries(retdec-utils shlwapi) # shlwapi.dll for PathRemoveFileSpec()
endif()
````
and  replace ````if(MSYS)```` to ````if(MINGW)````

Reply:
Thank you for the comment @Alexpux.

Reply:
Thank you all for the feedback. Will make the changes as suggested and test.